### PR TITLE
WIP: Start at updating packages. Not ready for prime time

### PR DESCRIPTION
**Pull request by spencerlyon2**
Warning: It is **NOT** ready to be merged

This is the start of some code that will allow us to update packages. I started on an async version, but didn't finish it. So I quickly wrote up a sync one. Any help to clean this up and get it working would be appreciated.

### CODE REVIEW

File: src/cljs/proton/lib/package_manager.cljs [hunk 1]
```diff
@@ -114,3 +114,57 @@
         merged-chan (async/merge chans)
         reduced-chan (async/reduce #(and %1 %2) true merged-chan)]
     reduced-chan))
+
+(defn outdated-packages-chan []
```
Comment (dvcrn) [inline src/cljs/proton/lib/package_manager.cljs]: I'd rename this to `get-outdated-packages` or something like that

**Comment by dvcrn** [comment]
I think we should force update proton on every start (if a new version is available) to make sure that the user always runs the latest code.

**Comment by spencerlyon2** [comment]
Given how long atom takes to start as it is, I think forcing that to happen _and_ having packages update would detract from the experience.

I like that spacemacs simply has a status bar notification that there are updates available. We have this too for free :)

That being said, the updating really wouldn't happen every time we start atom and it should only ever be a couple packages at a time. Can `core/async` provide a way for us to check if updates are available in a non-blocking way?

**Comment by dvcrn** [comment]
core.async is by nature non-blocking. Check out [this](http://swannodette.github.io/2013/08/02/100000-processes/) for example.

Maybe we can do a background upgrading of the proton-core (after the startup) and leave the rest optional?

**Comment by spencerlyon2** [comment]
That'd be really cool. It might get a little wacky if, say, your syntax or ui theme updates while using. Hopefully that doesn't happen often?

**Comment by dvcrn** [comment]
I forgot that I still had this laying around. I think instead of doing something separately, we can just integrate a auto-updater-package (like I just did). What do you think?

**Comment by spencerlyon2** [comment]
Sounds good to me. No need to re-invent the wheel here.

**Pull request closed without merging**